### PR TITLE
[WIP] Move git logic to Git Manager 

### PR DESCRIPTION
**Pull request by josephsawaya**
Signed-off-by: Joseph <jsawaya@redhat.com>

### CODE REVIEW

File: pkg/engine/fetchit.go [hunk 1]
```diff
@@ -721,63 +687,71 @@ func (hc *FetchitConfig) applyInitial(ctx context.Context, mo *SingleMethodObj,
...
+	}
+
+	changes, err := hc.gm.GetDiff(mo.Target.Name, working, latest)
 	if err != nil {
-		return utils.WrapErr(err, "error method: %s commit: %s", mo.Method, lc.Hash.String())
+		return err
 	}
 
-	hc.setlastCommit(mo.Target, mo.Method, newCommit)
 	hc.update(mo.Target)
```
Comment (sallyom) [inline pkg/engine/fetchit.go:726]: this update should be after you set the workingCommit, below, after this line: https://github.com/containers/fetchit/pull/177/files#diff-f909bb48f0bef398a8f323f6ebb382f1c1b48763963d34b5e90dfe6289b87b97R754

**Review by sallyom — CHANGES_REQUESTED** [review]
see the note about the placement of hc.update, other than that this lgtm

**Comment by sallyom** [comment]
I'm cool to merge this, wdyt? 
/lgtm

**Comment by josephsawaya** [comment]
Going to close this for now and come back to it later actually

**Pull request closed without merging**